Spigot-Server-Patches/0446-UUID-support-in-CraftProfileBanList.patch

```diff
@@ -0,0 +1,85 @@
+From 95b343c1556c2acfc47a9102f8bfe6cb676a0a88 Mon Sep 17 00:00:00 2001
+From: pop4959 <pop4959@gmail.com>
+Date: Wed, 16 Oct 2019 18:59:15 -0700
+Subject: [PATCH] UUID support in CraftProfileBanList
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftProfileBanList.java b/src/main/java/org/bukkit/craftbukkit/CraftProfileBanList.java
+index 84a1f9c8..6494029c 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftProfileBanList.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftProfileBanList.java
+@@ -14,6 +14,7 @@ import org.apache.commons.lang.Validate;
+ 
+ import com.google.common.collect.ImmutableSet;
+ import com.mojang.authlib.GameProfile;
++import java.util.UUID; // Paper
+ import java.util.logging.Level;
+ import org.bukkit.Bukkit;
+ 
+@@ -28,7 +29,14 @@ public class CraftProfileBanList implements org.bukkit.BanList {
+     public org.bukkit.BanEntry getBanEntry(String target) {
+         Validate.notNull(target, "Target cannot be null");
+ 
+-        GameProfile profile = MinecraftServer.getServer().getUserCache().getProfile(target);
++        // Paper start
++        GameProfile profile;
++        try {
++            profile = MinecraftServer.getServer().getUserCache().getProfile(UUID.fromString(target));
++        } catch (IllegalArgumentException e) {
++            profile = MinecraftServer.getServer().getUserCache().getProfile(target);
++        }
++        // Paper end
```

Comment:
I was less than thrilled as well with the implementation, but I didn't find another method that easily validates the UUID. Also, I was originally looking into whether this should go into it's own ban list type (such as a BanListType.UUID), or overloading the methods, but I thought that just results in way more code duplication (and a larger diff) since the underlying list is for profiles, and that won't change. I went with this because it seemed most effective to just make the existing methods work "better" in Paper but not diverge too much from the upstream interface. Let me know what your opinions are on this. I would definitely be fine with just adding overloaded methods if that is the recommended way to go about something like this.

Reply:
Creating a method `getBanEntry(UUID)` which calls `MinecraftServer.getServer().getUserCache().getProfile(theUuid)` directly is good enough honestly.

>just make the existing methods work "better" in Paper but not diverge too much from the upstream interface

If you worry about that, then you could try to PR to upstream instead.

Reply:
*ghh, misclicked enter*

>I didn't find another method that easily validates the UUID

There isn't any better way (iirc) indeed, but it's just ugly and feels dirty.

So in conclusion, you have two options:
1) Use overload with `UUID` parameter, it's a lot cleaner than that try/catch block - we don't mind having extra Paper-specific API here.
2) ^ still that, but PR to upstream if you don't want to diverge that much. If you have time and energy to wait/fight/whatever on upstream.